bin/proving-service/src/proxy/mod.rs

```diff
@@ -193,25 +195,18 @@ impl LoadBalancerState {
         self.workers.read().await.iter().filter(|w| !w.is_available()).count()
     }
 
-    /// Check the health of the workers and returns a list of healthy workers.
+    /// Check the status of the workers and update the health status.
     ///
-    /// Performs a health check on each worker using the gRPC health check protocol. If a worker
-    /// is not healthy, it won't be included in the list of healthy workers.
-    async fn check_workers_health(
-        &self,
-        workers: impl Iterator<Item = &mut Worker>,
-    ) -> Vec<Worker> {
-        let mut healthy_workers = Vec::new();
-
+    /// Performs a status check on each worker using the gRPC status protocol.
+    ///
+    /// If a worker is not ready, it will be marked as unhealthy and the number of unhealthy
+    /// workers will be incremented.
+    async fn check_workers_status(&self, workers: impl Iterator<Item = &mut Worker>) {
         for worker in workers {
-            if worker.is_healthy().await {
-                healthy_workers.push(worker.clone());
-            } else {
-                warn!("Worker {} is not healthy", worker.address());
+            if let Some(worker_status) = worker.check_status(&self.supported_prover_type).await {
+                worker.set_health_status(worker_status);
             }
         }
```

Comment:
Why do we use this pattern here? Why not set health status inside `Worker::check_status()`? Then, this could look something like this:
```rust
for worker in workers {
   worker.check_status(self.supported_prover_type).await;
}
```